iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/ProjectResourceFactory.java

```diff
@@ -41,6 +41,19 @@ private ProjectResourceFactory() {
     // utility class
   }
 
+  @CheckForNull
+  public static ProjectResource createResource(MappingTree tree) {
+    var kind = PropertyUtils.value(tree, "kind")
+      .map(ScalarTree.class::cast)
+      .map(ScalarTree::value)
+      .orElse("");
+    return switch (kind) {
+      case "ServiceAccount" -> createServiceAccount(tree);
+      case "LimitRange" -> createLimitRange(tree);
+      default -> null;
+    };
+  }
+
   @CheckForNull
   public static ProjectResource createServiceAccount(MappingTree tree) {
```

Comment:
If we access the Factory only by the `createResource()` method, we can remove the `public` modifier and drop the first condition.